include/xtensor/xtensor.hpp

```diff
@@ -548,6 +550,76 @@ namespace xt
 
         return out;
     };
+
+    /**
+     * Converts ``std::vector<index_type>`` (returned e.g. from ``xt::argwhere``) to a flattened
+     * ``xtensor``.
+     *
+     * @param vector of indices
+     *
+     * @return ``xt::xtensor<typename index_type::value_type, 1>`` (e.g. ``xt::xtensor<size_t, 1>``)
+     */
+    template <class T>
+    inline auto flatten_indices(const std::vector<T> &idx)
+    {
+        using return_type = xtensor<typename T::value_type, 1>;
+
+        auto n = idx.size();
+
+        if (n != 0)
+        {
+            n *= idx[0].size();
+        }
+
+        return_type out = empty<typename T::value_type>({n});
+        auto iter = out.begin();
+        for_each(idx.begin(), idx.end(), [&iter](const auto& t) { iter = std::copy(t.cbegin(), t.cend(), iter); });
+
+        return out;
+    };
+
+    /**
+     * Converts ``std::vector<index_type>`` (returned e.g. from ``xt::argwhere``) to ``xtensor``
+     * whereby the indices are ravelled. For 1-d input there is no conversion.
+     *
+     * @param idx vector of indices
+     * @param shape the shape of the original array
+     * @param l the layout type (row-major or column-major)
+     *
+     * @return ``xt::xtensor<typename index_type::value_type, 1>`` (e.g. ``xt::xtensor<size_t, 1>``)
+     */
+    template <class T, class S>
+    inline auto ravel_indices(const std::vector<T> &idx, const S& shape, layout_type l=layout_type::row_major)
```

Comment:
The code of this method is roughly equivalent to the one in `xstrides.hpp` so I think one of them can be removed. Regarding the return type, we could specify it as an additional template parameter with a default value. Since it depends on `T`, if we want  to put it as first parameter (so we don't have to specify other template parameters when we want to specify a return type different from the default one), we need to add an extra indirection:

```cpp
struct ravel_vector_tag;
struct ravel_tensor_tag;

template <class C, class Tag>
struct ravel_return_type;

template <class C>
struct ravel_return_type<C, ravel_vector_tag>
{
    using index_type = typename C::value_type;
    using value_type = typename index_type::value_type;
    using type = std::vector<vlaue_type>;
};

template <class C>
struct ravel_return_type<C, ravel_tensor_tag>
{
    using index_type = typename C::value_type;
    using value_type = typename index_type::value_type;
    using type = xt::xtensor<vlaue_type, 1>;
};

template <class C, class Tag>
using ravel_return_type_t = typename ravel_return_type<C, Tag>::type;

template <class Tag = ravel_tensor_tag, class C, class S>
ravel_return_type_t<C, Tag> ravel_indices(const C& idx, const S& shape, layout_type l = layout_type::row_major)
{
//....
}
``` 

Now if someone wants to get a `std::vector` instead of an `xtensor` he can simply write:
```cpp
auto v = ravel_indices<ravel_vector_flag>(...);
```

Reply:
That makes a lot of sense. 

(Can you think of a reason that people might want to use a `std::vector`? I only included it because it seems to be in line with the index vector that is returned from `xt::argwhere`. I.e. we could also not allow the `std::vector` option right now?)

Reply:
If they want to dynamically build a vector of indices. For instance, they compute indices with this method and then they want to `push_back` some indices precomputed from elsewhere.

Reply:
How can I construct generically for `std::vector` and `xt::xtensor` (to `idx.size()`)?

Reply:
I'm not sure you can do it (mainly because  `xtensor` is expecting a shape); you can add a static method to `ravel_return_type` and use it to initialize the `std::vector` / `xt::xtensor`.